apps/devhub/tasks.py

```diff
@@ -60,14 +60,14 @@ def validate(file_, listed=None, subtask=None):
 
 def validate_and_submit(addon, file_, listed=None):
     return validate(file_, listed=listed,
-                    subtask=submit_file.si(addon.pk, file_.pk))
+                    subtask=submit_file.si(addon.pk, file_.pk, listed=listed))
 
 
 @task
-def submit_file(addon_pk, file_pk):
+def submit_file(addon_pk, file_pk, listed):
     addon = Addon.unfiltered.get(pk=addon_pk)
     file_ = FileUpload.objects.get(pk=file_pk)
-    if (file_.passed_all_validations and
+    if ((not listed or file_.passed_all_validations) and
```

Comment:
this could just be `if ((addon.automated_signing or ...`, right? Same thing: https://github.com/mozilla/olympia/blob/master/apps/addons/models.py#L626 However, I dislike the name `automated_signing`